slyd/slyd/resources/annotations.py

```diff
@@ -1,216 +1,88 @@
-        field_id = None
-    return annotation_id, field_id
-
-
-def _load_relationships(attributes):
-    """Load relationships from jsonapi data. For field_id and container_id."""
-    relationships = {}
-    for key, value in attributes['relationships'].items():
-        relationships[key] = value.get('data') or {}
-        if hasattr(relationships[key], 'get'):
-            relationships['%s_id' % key] = relationships[key].get('id')
-    return relationships
-
-
-def _create_annotation(sample, attributes):
-    relationships = _load_relationships(attributes['data'])
-    attributes = attributes['data']['attributes']
-    annotations = sample['plugins']['annotations-plugin']['extracts']
-    aid = gen_id(disallow=[a['id'] for a in annotations if a.get('id')])
-    _id = gen_id(disallow={i for a in annotations for i in a.get('data', [])})
-    annotation = {
-        'id': aid,
-        'container_id': relationships['parent_id'].split('|')[0],
-        # TODO: default to most likely attribute
-        'data': {
-            _id: {
-                'attribute': attributes.get('attribute', 'content'),
-                'field': relationships['field_id'],
-                'required': False,
-                'extractors': []
-            }
-        },
-        'accept_selectors': attributes.get('accept_selectors', []),
-        'reject_selectors': attributes.get('reject_selectors', []),
-        'selector': attributes.get('selector'),
-        'required': [],
-        'tagid': attributes.get('tagid', '1')
-    }
-    if annotation['selector'] is None:
-        annotation['selector'] = ', '.join(annotation['accept_selectors'])
-    annotations.append(annotation)
-    return annotation
-
-
-def _find_schema_id(annotation, sample):
-    annotations = dict((a['id'], a) for a in sample['plugins']['annotations-plugin']['extracts'])
-    while annotation and 'schema_id' not in annotation:
-        container_id = annotation.get('container_id')
-        annotation = annotations.get(container_id)
-    return annotation and annotation.get('schema_id')
-
-
-def _create_field_for_annotation(manager, annotation, sample):
-    schema_id = _find_schema_id(annotation, sample)
-    field = create_field(manager, schema_id,
-                         {'data': {'attributes': {'type': 'text'},
-                          'type': 'fields'}})
-    field_id = field['data']['id']
-    for anno in annotation['data'].values():
-        if anno['field'] is None:
-            anno['field'] = field_id
-    return field['data']
+from operator import attrgetter
+
+from .route import (JsonApiRoute, ListModelMixin, RetrieveModelMixin,
+                    CreateModelMixin, UpdateModelMixin, DestroyModelMixin)
+from .serializers import AnnotationSchema
+from ..orm.base import AUTO_PK
+from ..orm.models import Project, Field, Item
+
+
+class AnnotationRoute(JsonApiRoute, ListModelMixin, RetrieveModelMixin,
+                      CreateModelMixin, UpdateModelMixin, DestroyModelMixin):
```

Comment:
The `AnnotationRoute`, `SchemaRoute`, `ExtractorRoute` and others have this list of inheritance/mixins. We could make an aggregate class which captures these classes which go together:
```
class RestfulRoute(JsonApiRoute, ListModelMixin, RetrieveModelMixin,
                   CreateModelMixin, UpdateModelMixin, DestroyModelMixin):
    pass
```